Sources/web3swift/Utils/EIP/EIP712.swift

```diff
@@ -20,28 +22,42 @@ public struct SafeTx: EIP712Hashable {
     let gasToken:       EIP712.Address
     let refundReceiver: EIP712.Address
     let nonce:          EIP712.UInt256
+
+    public init(to: EIP712.Address, value: EIP712.UInt256, data: EIP712.Bytes, operation: EIP712.UInt8, safeTxGas: EIP712.UInt256, baseGas: EIP712.UInt256, gasPrice: EIP712.UInt256, gasToken: EIP712.Address, refundReceiver: EIP712.Address, nonce: EIP712.UInt256) {
+        self.to = to
+        self.value = value
+        self.data = data
+        self.operation = operation
+        self.safeTxGas = safeTxGas
+        self.baseGas = baseGas
+        self.gasPrice = gasPrice
+        self.gasToken = gasToken
+        self.refundReceiver = refundReceiver
+        self.nonce = nonce
+    }
+
 }
 
 /// Protocol defines EIP712 struct encoding
-protocol EIP712Hashable {
+public protocol EIP712Hashable {
     var typehash: Data { get }
     func hash() throws -> Data
 }
 
-class EIP712 {
-    typealias Address = EthereumAddress
-    typealias UInt256 = BigUInt
-    typealias UInt8 = Swift.UInt8
-    typealias Bytes = Data
+public class EIP712 {
+    public typealias Address = EthereumAddress
+    public typealias UInt256 = BigUInt
+    public typealias UInt8 = Swift.UInt8
+    public typealias Bytes = Data
```

Comment:
Seems like it's redundant `public`

Reply:
they need to be marked as such, otherwise they have internal access, then the `SafeTX` intializer fails with `Initializer cannot be declared public because its parameter uses an internal type`

Reply:
Just to clarify, they are declared inside a `class` and even though the class is `public` that does not automatically apply to members of the class. If they were inside a `struct` the declarations would not be required.